operator/pkg/types/manifestResolver.go

```diff
@@ -0,0 +1,15 @@
+package types
+
+type ManifestResolver interface {
```

Comment:
I have really big troubles understanding what these methods are supposed to do without documentation and documenting which structs implement these interfaces (or could implement these). I think this is something we will have to invest in later if we really want people to use these abstractions.

Reply:
The entry point for a template-operator consumer is options, I added a comment there. 
I will also add a comment here.